code/modules/surgery/stomachpump.dm

```diff
@@ -39,7 +39,7 @@
 		display_results(user, target, "<span class='notice'>[user] forces [H] to vomit, cleansing their stomach of some chemicals!</span>",
 				"<span class='notice'>[user] forces [H] to vomit, cleansing their stomach of some chemicals!</span>",
 				"[user] forces [H] to vomit!")
-		H.vomit(20, FALSE, TRUE, 1, TRUE, FALSE, purge = TRUE) //called with purge as true to lose more reagents
+		H.vomit(20, FALSE, TRUE, 1, TRUE, FALSE, purge_ratio = 0.67) //higher purge ratio than regular vomiting
```

Comment:
Are you making a larger vomit change here?

Reply:
Yeah, i'll document it in the description. Functionality is unchanged, but now you can set a custom purge ratio (i.e. the percentage of reagents removed and put into the vomit). I refactored it so i had the option of giving a 0 purge ratio, instead of a binary 0.1 or 0.67. It might come in handy if you want to rebalance individual vomit purging methods in the future.